src/SearchResult.tsx

```diff
@@ -1,71 +1,115 @@
 import React, { useState } from "react";
-import Hightlighter from "react-highlight-words";
+import {
+  FaWindows,
+  FaLinux,
+  FaRegQuestionCircle,
+  FaLock,
+  FaLockOpen,
+  FaCircle,
+  FaRegCircle
+} from "react-icons/fa";
 
 interface SearchResultProps {
-  filename: string;
-  results: Array<any>;
-  openFunc: Function;
-  query: string;
+  name: string;
+  owner: string;
+  os: string;
+  isLocked: boolean;
+  isPoweredOn: boolean;
+  addMachineFunction: Function;
 }
 
 const SearchResult: React.SFC<SearchResultProps> = ({
-  filename,
-  results,
-  openFunc,
-  query
+  name,
+  owner,
+  os,
+  isLocked,
+  isPoweredOn,
+  addMachineFunction
 }) => {
-  const [isCollapsed, setIsCollapsed] = useState(false);
+  const getOsLogoComponent = () => {
+    // TODO: Change to switch case
+    if (os.toLowerCase() === "windows") {
+      return <FaWindows />;
+    } else if (os.toLowerCase() === "linux") {
+      return <FaLinux />;
+    } else {
+      return <FaRegQuestionCircle />;
+    }
+  };
+
+  const getLockLogoComponent = () => {
+    if (isLocked) {
```

Comment:
maybe one line:
```suggestion
    isLocked ? <FaLock /> : <FaLockOpen />
```

Same for `getPowerLogoComponent`

Reply:
👍